addons/test_mail/tests/test_mail_mail.py

```diff
@@ -728,6 +733,86 @@ def test_mail_mail_send_server(self):
         self.assert_email_sent_smtp(message_from='user_1@test_2.com', emails_count=5, from_filter=self.server_domain.from_filter)
 
 
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mail_mail_send_server_attachment_to_download_link(self):
+        """ Test that when the mail size exceeds the max email size limit,
+        attachments are turned into download links added at the end of the
+        email content.
+
+        The test uses the low level method _send of mail_mail to perform the tests:
+        - using a specified server or the default one (to test command line parameters)
+        - sending one or more than one mail at once
+        - with mail that exceed (with one or more attachments) or not the limit
+        """
+        def count_attachments(message):
+            return (0 if isinstance(message, str)
+                    else sum(count_attachments(part) for part in message.get_payload()) if message.is_multipart()
+                    else 'attachment' in message.get('Content-Disposition', ''))
+
+        config_get = config.get
+        attachment_size_in_bytes = 1024 * 128
+        datas = base64.b64encode((''.join('.' for _ in range(0, attachment_size_in_bytes))).encode())
+
+        # max_email_size_inv produces the opposite of what is expected, it is assigned to the mail server not used
+        for idx_test, ((max_email_size, max_email_size_inv, turned_into_links, n_attachment), mail_server, n_mail) in \
```

Comment:
We generally try to use "expected" in variable naming for expected output, to know it is not an input (like 'expected_is_convert', 'expected_attach_count', ...)